src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_base.erl

```diff
@@ -870,11 +884,53 @@ api_info_header_value(#base_state{}=State) ->
                     {Key, Value} <- api_info(State)],
                 ";").
 
-%% @doc Add the X-Ops-API-Info header to the outgoing response.  This contains server API
+%% @doc Add the X-Ops-Server-API-Version to the outgoing response. This will be of the format:
+%% {"min_version":"<non-negative integer>",
+%% "max_version":"<non-negative integer>",
+%% "request_version":"<integer>",
+%% "response_version":"<integer>"}
+%%
+%% min_version and max_version:
+%%   integers representing the range of API versions the server supports
+%% request_version:
+%%   an integer representing the desired API version from the client via X-Ops-Server-API-Version
+%%   or zero if none was sent. if X-Ops-Server-API-Version contained an invalid value (not an integer),
+%%   then -1 is returned.
+%% response_version:
+%%   an integer representing the API version used by the server to process the request.
+%%   it either matches what the client requested in X-Ops-Server-API-Version or is -1 if a 406 occurred.
+%%
+%% Deprecated: Add the X-Ops-API-Info header to the outgoing response.  This contains server API
 %% version information (useful for maintaining back-compatibility) as well as OTP version
 %% information (more useful for debugging purposes).
-add_api_info_header(Req, State) ->
-    wrq:set_resp_header("X-Ops-API-Info", api_info_header_value(State), Req).
+-spec add_api_info_header(#wm_reqdata{}, #base_state{}, atom()) ->
+                                 #wm_reqdata{}.
+add_api_info_header(Req, #base_state{server_api_version = APIVersion} = State, undefined) ->
+    Req0 = wrq:set_resp_header("X-Ops-API-Info", api_info_header_value(State), Req),
+    HeaderBody = build_api_info_header_body(APIVersion, APIVersion),
+    wrq:set_resp_header("X-Ops-Server-API-Version", HeaderBody, Req0);
+add_api_info_header(Req, #base_state{server_api_version = RequestedVersion} = State, not_acceptable) ->
+    %% indicates the request wasn't processed due to a 406
+    ActualAPIVersion = -1,
+    RequestedAPIVersion = case RequestedVersion of
```

Comment:
Er actually, the arity is 3 for line 914 actually. We want to respond with `-1`as the `response_version` in the 406 case, so I'm not sure how moving that `case` statement would help with that.